core/runtime/jsruntime.rs

```diff
@@ -1964,6 +1981,18 @@ impl JsRuntime {
     let mut context_state = context_state.borrow_mut();
     let mut dispatched_ops = false;
 
+    // Poll any pending task spawner tasks
+    // TODO(mmastrac): we are polling this here because we already have a lock on context_state
+    if let Poll::Ready(tasks) =
+      context_state.task_spawner_factory.poll_inner(cx)
+    {
+      // TODO(mmastrac): we are using this flag
+      dispatched_ops = true;
+      for task in tasks {
+        task(scope);
```

Comment:
From V8's API documentation for `ThrowException`:

> When an exception has been scheduled it is illegal to invoke any JavaScript operation; the caller must return immediately and only after the exception has been handled does it become legal to invoke JavaScript operations.

I believe that this also extends to using scopes in C++ / Rust. Effectively, since now this `task` can trigger any extension provided Rust code that can then do anything it wants with the scope we'll need to prepare for the eventuality that some code being called will throw an error.

If a JS error thrown in a task breaks the event loop then we're in a bad state.

Reply:
Interesting. How do we currently handle this in middleware? We may just need to copy the same requirements over to this task spawner.

Reply:
At least FFI middleware creates its own CallbackScope and then a HandleScope from that. If it throws it only poisons itself.